api/v1alpha1/flowcollector_types.go

```diff
@@ -201,6 +201,44 @@ type FlowCollectorKafka struct {
 	TLS ClientTLS `json:"tls"`
 }
 
+const (
+	PrometheusTLSDisabled = "DISABLED"
+	PrometheusTLSManual   = "MANUAL"
+	PrometheusTLSAuto     = "AUTO"
+)
+
+type PrometheusTLSConfigType string
+
+// PrometheusTLS define the TLS configuration of Prometheus
+type PrometheusTLS struct {
+	// Select the type of TLS configuration
+	// "DISABLED" (default) to not configure TLS for the endpoint, "MANUAL" to manually provide cert file and a key file,
+	// and "AUTO" to use Openshift auto generated certificate using annotations
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="DISABLED";"MANUAL";"AUTO"
+	// +kubebuilder:validation:Required
+	//+kubebuilder:default:="DISABLED"
+	Type PrometheusTLSConfigType `json:"type,omitempty"`
+
+	// TLS configuration.
+	// +optional
+	Manual *CertificateReference `json:"manual"`
```

Comment:
According to the recommendation this field name should match with the name in the enum discriminator.
I will also set it to provided to match your other comment.

Reply:
could it be `CERTIFICATE` is a valid option for the `tls` type field instead of `MANUAL` or `PROVIDED`?

Reply:
I think having only CERTIFICATE would be a bit ambiguous since AUTO mode also use certificate, it is just that they are generated and not provided by the user.